pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -230,37 +234,76 @@ func TestAmbientIndex_WaypointAddressAddedToWorkloads(t *testing.T) {
 
 	// Add pods for app "a".
 	s.addPods(t, "127.0.0.1", "pod1", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
+	s.assertEvent(t, s.podXdsName("pod1"))
 	s.addPods(t, "127.0.0.2", "pod2", "sa1", map[string]string{"app": "a", "other": "label"}, nil, true, corev1.PodRunning)
+	s.assertEvent(t, s.podXdsName("pod2"))
 	s.addPods(t, "127.0.0.3", "pod3", "sa1", map[string]string{"app": "a"}, nil, true, corev1.PodRunning)
+	s.assertEvent(t, s.podXdsName("pod3"))
+	// Add pods for app "b".
+	s.addPods(t, "127.0.0.4", "pod4", "sa2", map[string]string{"app": "b"}, nil, true, corev1.PodRunning)
+	s.assertEvent(t, s.podXdsName("pod4"))
+
+	// Prime gateway informer with throw away events
+	s.addWaypoint(t, "10.0.0.2", "primer", "")
+	s.deleteWaypoint(t, "primer")
 	s.clearEvents()
+	stop := test.NewStop(t)
+	s.controller.client.WaitForCacheSync("test", stop)
```

Comment:
delete this and add `CRDs:             []schema.GroupVersionResource{gvr.KubernetesGateway},` to L939 (and can remove MakeCRD)

Reply:
Put a hold for this

Reply:
Thanks for the suggestions! This seems like less of a kludge than having the "primer" events anyhow.  I'll give this a try.